addons/sale_stock/models/stock.py

```diff
@@ -38,11 +41,16 @@ def _get_related_invoices(self):
 
     def _assign_picking_post_process(self, new=False):
         super(StockMove, self)._assign_picking_post_process(new=new)
-        if new and self.sale_line_id and self.sale_line_id.order_id:
-            self.picking_id.message_post_with_view(
-                'mail.message_origin_link',
-                values={'self': self.picking_id, 'origin': self.sale_line_id.order_id},
-                subtype_id=self.env.ref('mail.mt_note').id)
+        if new:
+            moves = self.filtered(lambda m: m.sale_line_id)
+            sorted_moves = sorted(moves, key=lambda m: (m.sale_line_id.order_id.id, m.picking_id.id))
+            grouped_moves = groupby(sorted_moves, key=lambda m: (m.sale_line_id.order_id, m.picking_id))
```

Comment:
simplify all this
sale_orders = self.mapped('sale_line_id.order_id')
for sale_order in sale_orders..

no need to sort/groupby